update to boost 1.82

thomashampson opened the pull request:
### Description of work
Update boost to 1.82 in line with [conda-forge](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/604816c8db83f9ceefbc4a04ba4de09747cf3822/recipe/conda_build_config.yaml#L458).

Fixes #36687 

### To test
PR checks should provide adequate coverage.


<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

thomashampson committed:
update to boost 1.82

thomashampson committed:
add release notes

jhaigh0 commented:
Re the test failue in `LoadStlTest` on macos. The test is testing the `HasV3DPair` function which uses `boost::hash_combine()`.

On the boost changes [here](https://www.boost.org/doc/libs/1_82_0/libs/container_hash/doc/html/hash.html#recent_boost_1_78_0) it says **for boost 1.78 (aka the old version)**

_Fixed `hash_combine `so that its behavior no longer depends on whether `size_t `is the exact same type as `boost::uint64_t` (which wasn’t the case on macOS). This changes the hash values of composite (container and tuple) types on macOS._

So that seems pretty suspect but the change was for 1.78 which is what we were on already, so a bit confusing.

thomashampson committed:
Use hash_combine for all elements of V3D

This follows more closely the example in the boost documentation.

thomashampson committed:
Add override keyword to fix cppcheck warning

thomashampson committed:
Remove explicit boost runtime requirement from mantid recipes

The relevant boost runtime deps should be brought in by the run_exports
defined in the boost conda recipes.

robertapplin reviewed (commented):
Code changes look good, and all the PR checks are passing so far (just waiting on Windows). Happy to approve when this is marked as ready
